sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -0,0 +1,601 @@
+     * NOTE: this method does not respect message locks or increment delivery count
+     * for messages.
+     * @param fromSequenceNumber The sequence number to start peeking from (inclusive).
+     * @param maxMessageCount The maximum number of messages to retrieve.
+     */
+    peekBySequenceNumber(
+      fromSequenceNumber: Long,
+      maxMessageCount?: number
+    ): Promise<ReceivedMessage[]>;
+  };
+}
+
+/**
+ * Methods to manage rules for subscriptions. More information about subscription rules
+ * can be found here: https://docs.microsoft.com/en-us/azure/service-bus-messaging/topic-filters
+ */
+export interface SubscriptionRuleManagement {
+  /**
+   * Gets all rules associated with the subscription
+   * @throws Error if the SubscriptionClient or the underlying connection is closed.
+   * @throws MessagingError if the service returns an error while retrieving rules.
+   */
+  getRules(): Promise<RuleDescription[]>;
+
+  /**
+   * Removes the rule on the subscription identified by the given rule name.
+   *
+   * **Caution**: If all rules on a subscription are removed, then the subscription will not receive
+   * any more messages.
+   * @param ruleName
+   * @throws Error if the SubscriptionClient or the underlying connection is closed.
+   * @throws MessagingError if the service returns an error while removing rules.
+   */
+
+  removeRule(ruleName: string): Promise<void>;
+  /**
+   * Adds a rule on the subscription as defined by the given rule name, filter and action.
+   *
+   * **Note**: Remove the default true filter on the subscription before adding a rule.
+   * Otherwise, the added rule will have no affect as the true filter will always result in
+   * the subscription receiving all messages.
+   * @param ruleName Name of the rule
+   * @param filter A Boolean, SQL expression or a Correlation filter. For SQL Filter syntax, see
+   * {@link https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-sql-filter SQLFilter syntax}.
+   * @param sqlRuleActionExpression Action to perform if the message satisfies the filtering expression. For SQL Rule Action syntax,
+   * see {@link https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-sql-rule-action SQLRuleAction syntax}.
+   * @throws Error if the SubscriptionClient or the underlying connection is closed.
+   * @throws MessagingError if the service returns an error while adding rules.
+   */
+  addRule(
+    ruleName: string,
+    filter: boolean | string | CorrelationFilter,
+    sqlRuleActionExpression?: string
+  ): Promise<void>;
+  readonly defaultRuleName: string;
+}
+
+/**
+ * The Receiver class can be used to receive messages in a batch or by registering handlers.
+ * Use the `createReceiver` function on the QueueClient or SubscriptionClient to instantiate a Receiver.
+ * The Receiver class is an abstraction over the underlying AMQP receiver link.
```

Comment:
Outdated.

Reply:
Fixed (by removing it). Those classes aren't "public" anymore and the description on the interfaces should suffice.